state-chain/pallets/cf-broadcast/src/lib.rs

```diff
@@ -407,7 +407,9 @@ pub mod pallet {
 				let retries_len = retries.len();
 
 				for retry in retries {
-					Self::start_next_broadcast_attempt(retry);
+					if Self::take_awaiting_broadcast(retry.broadcast_attempt_id).is_some() {
```

Comment:
Is there still an issue here? 

Scenario:
1. We take this attempt and start next broadcast
2. Signature is invalid, so we re-threshold sign (and don't yet insert the new attempt)
4. Transaction success witnessed for the retried/timedout broadcast

In success we would do: 
```
AwaitingBroadcast::<T, I>::get(BroadcastAttemptId {
						broadcast_id,
						attempt_count: BroadcastAttemptCount::<T, I>::get(broadcast_id),
```
But we're taking this here.

Reply:
Will write a test to confirm this.